src/runtime_src/xocl/xclbin/xclbin.cpp

```diff
@@ -726,8 +726,8 @@ class xclbin_data_sections
         // first element not part of the sorted (decreasing) range
         auto upper = std::find_if(itr, m_membanks.end(), [addr] (auto& mb) { return mb.base_addr < addr; });
 
-        // find first used memidx if any, default to first memidx in range if unused
-        auto used = std::find_if(itr, upper, [](auto& mb) { return mb.used; });
+        // find last used memidx if any, default to first memidx in range if unused
+	auto used = std::find_end(itr, upper, itr, itr+1, [](auto& mb, auto& mb2) { (void)mb2; return mb.used;});
```

Comment:
Can you simplify this logic by getting "upper". Just get the "upper" value by checking both addr and size

Reply:
Updated the logic.